env0/resource_gcp_credentials.go

```diff
@@ -67,10 +72,15 @@ func resourceGcpCredentialsRead(ctx context.Context, d *schema.ResourceData, met
 	apiClient := meta.(client.ApiClientInterface)
 
 	id := d.Id()
-	_, err := apiClient.CloudCredentials(id)
+	credentials, err := apiClient.CloudCredentials(id)
 	if err != nil {
 		return ResourceGetFailure("gcp credentials", d, err)
 	}
+
+	if err := writeResourceData(&credentials, d); err != nil {
+		return diag.Errorf("schema resource data serialization failed: %v", err)
+	}
```

Comment:
Just asking, what changed that we now need it than before?
Is `import` flow will not work without it?

Reply:
@razbensimon ,

The writeResourceData (and readReasourceData) is a utility function that reads all the fields in the interface and updates the schema via `set` calls.
This is future proof: in case you add more fields to the schema/api - no additional coding is needed (other than updating the schema).

There is a potential bug here that drifting is not detected (E.g., if someone changes the name in the UX). Once I added this code, it caught a bug in the tests as well (so I fixed the tests as well).
Import flow will work without it, and I just fixed bugs I encountered on the way.